mindmeld/components/dialogue.py

```diff
@@ -848,10 +861,7 @@ def _validate(self, request, slot):
             # Custom validation using function provided by developer. Should return True/False
             # for validation status. If true, then continue, else fail overall validation.
 
-            if (
-                slot.custom_eval(text) not in (True, False)
-                or slot.custom_eval(text) is False
-            ):
+            if slot.custom_eval(text, request) is not True:
```

Comment:
Nit:
```suggestion
            if not slot.custom_eval(text, request):
```

Reply:
Can't use ``not``. Negation only applies to boolean cases. This is handling any output other than a True value (which could be non boolean too).

Reply:
That is not our problem, in the sense that the custom eval functions are expected to return bool anyway?

Reply:
It is, but then we'd need to do error handling separately. Isn't this a shorter fix?

Reply:
The docs (that you created! 😄) say `custom_eval` should return a boolean. I would just make that assumption here. I don't see why there should be any error handling to determine whether we received a boolean value when that's what the docs dictate. If someone misuses it, the behavior is undefined.

In any case, explicit comparisons to `True` or `False` are a style no-no. It implies the expression could be something _other_ than boolean, and mixing other types with boolean is an even bigger no-no.

Reply:
Okay! Makes sense :)